### PR TITLE
FIx Bark batching feature

### LINKED ISSUE #25861: Bark TTS - batch inference silence/noise
Hi, 
when I perform batch inference with Bark model, all the output `.wav` files have the same duration, even though some transcriptions are much shorter than others, and the difference is filled with silence/noise. Is there a way to get information about when the model stops synthesizing the transcription, so that I can trim that silence after the generation process?

### PR DESCRIPTION
**Pull request by ylacombe**
# What does this PR do?

This PR aims to fix batching with Bark:

> Currently, when several samples are transmitted to Bark at the same time, i.e. during batch generation, the generated audios all have the same duration, which extends the length of the shortest audios to the longest audio in the batch.
> This problem can be solved by keeping track of sample lengths as Bark.generate is called up, and then outputting the audio lengths with the generated audios at the end of the call.

This fix can be enabled with an additional bolean parameter (`return_output_lengths`, same naming than in #25943) so I've kept backward compatibility.
I also made sure that generated batched outputs were almost the same than without batching and listened to generated audio qualitatively!

cc @sanchit-gandhi and @amyeroberts, could you take a look ? Many thanks!

Fixes #25861 and #26673

### CODE REVIEW

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing this! 

Changes look OK to me. Having methods conditionally return different types and number of objects isn't ideal but looks like the simplest way to add this.

I'd like a second review from @gante before merging to confirm everything's OK with `generate` (although Bark's generate is atypical anyway).

**Review by gante — APPROVED** [review]
LGTM, thanks for fixing 👍

File: tests/models/bark/test_modeling_bark.py
Comment (ylacombe) [reply tests/models/bark/test_modeling_bark.py:1094]: No gold standard yet tbh !

Comment (sanchit-gandhi) [reply tests/models/bark/test_modeling_bark.py:1094]: (coming in late) 2e-3 is pretty good for a full-size generative audio model. It's hard to get much closer when you generate continuous outputs. For example, we use 1e-3 in `diffusers` for generative audio models: https://github.com/huggingface/diffusers/blob/0f55c17e1742aee3e34e419c6856d044746a1fd0/tests/pipelines/audioldm2/test_audioldm2.py#L569

**Comment by sourabharsh** [comment]
This was very helpful. Thanks!

**Pull request merged**
